packages/scene-composer/src/augmentations/components/three-fiber/viewpoint/ViewCursorWidget.tsx

```diff
@@ -1,76 +1,94 @@
+import { ViewCursorEditIcon, ViewCursorMoveIcon } from '../../../../assets';
+import { getIntersectionTransform } from '../../../../utils/raycastUtils';
 import { sceneComposerIdContext } from '../../../../common/sceneComposerIdContext';
 import { useEditorState } from '../../../../store';
 
-// Adds the custom objects to React Three Fiber
-extend({ ViewCursor, Viewpoint, WidgetVisual });
-
-const AsyncLoadViewCursorWidget: React.FC = () => {
+export const ViewCursorWidget = () => {
+  const { scene } = useThree();
+  const ref = useRef<Object3D>(null);
   const sceneComposerId = useContext(sceneComposerIdContext);
-  const { cursorPosition, cursorLookAt, cursorVisible, cursorStyle } = useEditorState(sceneComposerId);
-  const viewCursorRef = useRef<Viewpoint>(null);
-  const { camera } = useThree();
+  const { cursorVisible, cursorStyle } = useEditorState(sceneComposerId);
 
-  const moveVisual = useMemo(() => {
-    const svgData: SVGResult = useLoader(SVGLoader, ViewCursorMoveIcon.dataUri);
-    return svgIconToWidgetVisual(svgData, ViewpointState.Deselected, false, { userData: { isCursor: true } });
-  }, []);
+  const svg = cursorStyle === 'edit' ? ViewCursorEditIcon : ViewCursorMoveIcon;
+  const data = useLoader(SVGLoader, svg.dataUri);
 
-  const editVisual = useMemo(() => {
-    const svgData: SVGResult = useLoader(SVGLoader, ViewCursorEditIcon.dataUri);
-    return svgIconToWidgetVisual(svgData, ViewpointState.Selected, false, { userData: { isCursor: true } });
-  }, []);
+  /* istanbul ignore next */
+  const resetObjectCenter = useCallback((object: Object3D) => {
+    const box = new Box3().setFromObject(object);
+    box.getCenter(object.position);
+    object.position.multiplyScalar(-1);
+  }, [ref]);
 
-  useEffect(() => {
-    if (viewCursorRef.current) {
-      viewCursorRef.current.lookAt(cursorLookAt);
-      viewCursorRef.current.rotation.z = camera.rotation.z; // Prevent spinning and always be straight up and down
-    }
-  }, [cursorLookAt]);
+  /* istanbul ignore next */
+  const shape = useMemo(() => {
+    const paths = data.paths;
+    const svgGroup = new Group();
 
-  useEffect(() => {
-    if (viewCursorRef.current) {
-      viewCursorRef.current.traverse((child) => {
-        const mesh = child as THREE.Mesh;
+    paths?.forEach((path) => {
+      const fillColor = path?.userData?.style.fill;
+      if (fillColor !== undefined && fillColor !== 'none') {
+        const material = new MeshBasicMaterial({
+          color: new Color().setStyle(fillColor).convertSRGBToLinear(),
+          opacity: path?.userData?.style.fillOpacity,
+          transparent: true,
+          depthFunc: AlwaysDepth,
+          side: DoubleSide
+        });
+        const shapes = SVGLoader.createShapes(path);
+        shapes.forEach((line) => {
+          const geometry = new ShapeGeometry(line);
+          const mesh = new Mesh(geometry, material);
+          resetObjectCenter(mesh);
+          svgGroup.add(mesh);
+        });
+      }
```

Comment:
This logic still feels out of place here, moving to utility functions that break up some of this complexity could make both this Component and this logic easier to test.

Reply:
Great point.